### PR TITLE
Peer review

### PR DESCRIPTION
**Pull request by oahray**
Please leave feedback.

### CODE REVIEW

File: client/src/containers/Group.jsx [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+      selectedMessage: {}
+    });
+  }
+
+  newMessageAdded(groupid) {
+    return this.props.getGroupMessages(groupid, this.props.token);
+  }
+
+  scrollToBottom() {
+    // this.messageList.scrollTop = this.messageList.scrollHeight;
```
Comment (johnkingzy) [inline client/src/containers/Group.jsx]: All lines within method has benn commented out, can you take this off

File: client/src/actions/auth/requestReset.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  };
+};
+
+export const clearResetRequestMessage = () => {
+  return {
+    type: CLEAR_REQUEST_RESET_MESSAGE
+  };
+};
+
+export const requestReset = (email) => {
```
Comment (veryobinna) [inline client/src/actions/auth/requestReset.js]: You need to add JSDocs to all functions.

File: client/src/actions/auth/signinAction.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+const signinFailure = (error) => {
+  return {
+    type: SIGNIN_FAILURE,
+    error
+  };
+};
+
+export const signinUser = (username, password) => {
+  return (dispatch) => {
```
Comment (veryobinna) [inline client/src/actions/auth/signinAction.js]: You're passing dispatch to the anonymous function, you seem not to make use of the argument.

File: client/src/actions/auth/signupAction.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      }
+    })
+    .then((response) => {
+      if (response.statusText === 'Created') {
+        dispatch(signupSuccess(response));
+      }
+    })
+    .catch((err) => {
+      if (err.response) {
+        console.log(err.response.data);
```
Comment (veryobinna) [inline client/src/actions/auth/signupAction.js]: No console logs

File: client/src/actions/group/getGroupsList.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+const groupsListFailure = (response) => {
+  return {
+    type: GROUPS_LIST_FAILURE,
+    response
+  };
+};
+
+export const getGroupList = (token) => {
+  const FETCH_URL = `${BASE_URL}/user/me/groups`;
+  return (dispatch, getState) => {
```
Comment (veryobinna) [inline client/src/actions/group/getGroupsList.js]: are the arguments necessary?

File: client/src/actions/group/searchUsers.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    .catch((err) => {
+      if (err.response) {
+        dispatch(searchUsersFailure(err.response.data.error));
+      }
+    });
+  };
+};
+
+
+// ADD USER FROM SEARCH RESUKTS
```
Comment (veryobinna) [inline client/src/actions/group/searchUsers.js]: You can do better docs with JSDocs

File: client/src/actions/group/searchUsers.js [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+};
+
+const addUserFailure = (error) => {
+  return {
+    type: ADD_USER_FAILURE,
+    error
+  };
+};
+
+// export const resetAddUserStatus = () => {
```
Comment (veryobinna) [inline client/src/actions/group/searchUsers.js]: Remove unnecessary code

File: client/src/containers/App.jsx [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+import { verifyAuth } from '../actions';
+import Preloader from '../components/Preloader';
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  componentWillMount() {
+    if (window.localStorage && window.localStorage.getItem('x-auth')) {
```
Comment (veryobinna) [inline client/src/containers/App.jsx]: window.localstorage.getItem() is fine. You can remove the window.localstorage.

File: client/src/containers/Group.jsx [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+    socket.on('Added to group', ({ group }) => {
+      if (group.id === this.props.selectedGroup.id) {
+        this.props.getGroupUsers(this.groupId, this.props.token);
+      }
+    });
+  }
+
+  componentWillReceiveProps(newProps) {
+    if (this.props.selectedGroup && Number(newProps.match.params.groupid) !== this.props.selectedGroup.id) {
+      console.log(typeof newProps.match.params.groupid);
```
Comment (veryobinna) [inline client/src/containers/Group.jsx]: no console logs

File: client/src/containers/Group.jsx [hunk 10]
```diff
@@ -0,0 +1,262 @@
...
+      <div className="row col s12">
+        <div className="tab-content col s12">
+          <Message message={this.state.selectedMessage} 
+          closeMessage={this.closeMessage} user={this.props.user}/>
+        </div>
+      </div>
+    );
+
+    const messagesTab = (
+      this.state.messageOpen ? showOpenMessage : messages
```
Comment (veryobinna) [inline client/src/containers/Group.jsx:390]: don't set state directly, use the setState method

File: client/src/containers/UserHome.jsx [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+  return {
+    isLoggedIn: state.isAuthenticated,
+    user: state.user,
+    groupList: state.groupList,
+    groupListLoading: state.groupListLoading,
+    token: state.token
+  }
+}
+
+function mapDispatchToProps(dispatch) {
```
Comment (veryobinna) [inline client/src/containers/UserHome.jsx]: all functions should be in es6

File: client/src/reducers/createdGroup.js [hunk 12]
```diff
@@ -0,0 +1,12 @@
+import { CREATE_GROUP_SUCCESS ,RESET_CREATE_GROUP_STATUS } from '../actions';
+
+export default (state = null, action) => {
+  switch (action.type) {
+    case CREATE_GROUP_SUCCESS:
```
Comment (veryobinna) [inline client/src/reducers/createdGroup.js]: Don't you think state will be mutated here? Make use of es6 spread operator to add new state.

Comment (veryobinna) [reply client/src/reducers/createdGroup.js]: Do the  same for all reducers

File: coverage/bc-24-postit/server/models/group.js.html [hunk 13]
```diff
@@ -0,0 +1,164 @@
+<!doctype html>
```
Comment (veryobinna) [inline coverage/bc-24-postit/server/models/group.js.html]: Why do you have HTML on the backend path?

File: server/routes/index.js [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+  app.delete('/api/group/:groupid/user', authenticate, isValidUsername,
+  isGroupMember, groupsController.removeUserFromGroup);
+
+  app.post('/api/group/:groupid/leave', authenticate,
+  isGroupMember, groupsController.leaveGroup);
+
+  app.get('/api/group/:groupid/users', authenticate,
+  isGroupMember, groupsController.getGroupUsers);
+
+  app.post('/api/group/:groupid/message', authenticate,
```
Comment (johnkingzy) [inline server/routes/index.js]: I see you are currently using express v4.x which provides you with a router instance to chain endpoint of different methods. Use this as a guide 

https://www.codementor.io/johnkingzy/api-routing-using-expressjs-router-4-0-9yohet6uu

File: server/middleware/isTaken.js [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+  const username = req.body.username.trim().toLowerCase();
+  const email = req.body.email.trim().toLowerCase();
+  User.findOne({ where: { username } }).then((user) => {
+    if (user) {
+      return res.status(400).send({
+        error: 'Username already taken.'
+      });
+    }
+    User.findOne({ where: { email } }).then((userAgain) => {
+      if (userAgain) {
```
Comment (johnkingzy) [inline server/middleware/isTaken.js]: What do you mean by user again?

File: server/middleware/isGroupMember.js [hunk 16]
```diff
@@ -0,0 +1,34 @@
+import { Group } from '../models';
+
+export default (req, res, next) => {
+  const groupId = req.params.groupid;
+  if (!Number(groupId)) {
```
Comment (johnkingzy) [inline server/middleware/isGroupMember.js]: Instead of using the Javascript Number Object, try `isNaN()`

File: server/controllers/users.js [hunk 17]
```diff
@@ -0,0 +1,296 @@
+import * as _ from 'lodash';
+import randomstring from 'randomstring';
+import { User, Message } from '../models';
+import { transporter, helperOptions } from '../config/nodemailer';
+
+// Function to signup new users
+export const signup = (req, res) => {
+  const username = req.body.username.trim().toLowerCase();
+  const email = req.body.email.trim().toLowerCase();
```
Comment (johnkingzy) [inline server/controllers/users.js]: What if I pass in an invalid email address for example `emailaddress`? I can't see where you covered that scenario.

File: server/controllers/users.js [hunk 18]
```diff
@@ -0,0 +1,296 @@
...
+  .catch(() => res.status(400));
+};
+
+// Function to sign users in
+export const signin = (req, res) => {
+  const body = _.pick(req.body, ['username', 'password']);
+  const username = body.username.trim().toLowerCase();
+  if (!body.password) {
+    return res.status(400).json({
+      error: 'Password must not be empty'
```
Comment (johnkingzy) [inline server/controllers/users.js]: I will suggest you try as much as possible to be consistent with your design pattern, I can see you are using a middleware to validate the username, but you seem to be validating the password inside controller.

File: server/controllers/users.js [hunk 19]
```diff
@@ -0,0 +1,296 @@
+import * as _ from 'lodash';
+import randomstring from 'randomstring';
+import { User, Message } from '../models';
```
Comment (johnkingzy) [inline server/controllers/users.js:3]: Leave a blank line between relative imports and absolute imports

File: server/app.js [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+import webpackHotMiddleware from 'webpack-hot-middleware';
+import webpackConfig from '../webpack.config';
+
+// Set up the express app and middleware
+const app = express();
+
+const compiler = webpack(webpackConfig);
+
+const env = process.env.NODE_ENV || 'development';
+console.log('>>>>>> ENV: ', env);
```
Comment (johnkingzy) [inline server/app.js]: I feel you might want to take this out, `console.log`

File: server/app.js [hunk 21]
```diff
@@ -0,0 +1,75 @@
...
+
+dotenv.config();
+
+if (env === 'development') {
+  dotenv.config();
```
Comment (johnkingzy) [inline server/app.js]: i don't think you need to repeat `dotenv.config()` twice.

File: server/app.js [hunk 22]
```diff
@@ -0,0 +1,75 @@
...
+};
+app.use(cors(corsOptions));
+
+const publicPath = path.join(__dirname, '../client/public/');
+const indexPath = path.resolve(__dirname, publicPath, 'index.html');
+
+app.use('/', express.static(publicPath));
+
+// Require our routes
+require('./routes')(app);
```
Comment (johnkingzy) [inline server/app.js:68]: I will suggest you are consistent your with coding style, I can see you are using es6 module system which is `import/export` but right here you happen to be using the commonjs module system which is `require` here.

File: server/config/config.js [hunk 23]
```diff
@@ -0,0 +1,27 @@
...
+    password: process.env.DB_PASSWORD,
+    database: 'postit-test',
+    host: '127.0.0.1',
+    port: 5432,
+    dialect: 'postgres',
+    logging: false
+  },
+  production: {
+    username: 'tpgpwvuylmgwfq',
+    password: '721df25847ce113ee7306f1c673d092b788b53ea48ad8b75f797c25695f912b8',
```
Comment (johnkingzy) [inline server/config/config.js]: your private info is visible, try securing this as well.

File: client/src/reducers/setUserReducer.js [hunk 24]
```diff
@@ -0,0 +1,20 @@
+import { VERIFY_AUTH_SUCCESS, SIGNUP_SUCCESS,
+  SIGNIN_SUCCESS, LOGOUT_USER, RESET_PASSWORD_SUCCESS } from '../actions';
+
+export default function (state = null, action) {
+  console.log('>>>>>>>>>>>>>> User Reducer Triggered! by:', action);
```
Comment (johnkingzy) [inline client/src/reducers/setUserReducer.js]: I think you should take this off

File: client/src/containers/UserHome.jsx [hunk 25]
```diff
@@ -0,0 +1,80 @@
...
+
+    return (
+      <div>
+        {this.props.isLoggedIn ? content : <Redirect to="/" />}
+      </div>
+    )
+  }
+}
+
+function mapStateToProps(state) {
```
Comment (johnkingzy) [inline client/src/containers/UserHome.jsx]: Missing JSDoc for function

**Pull request closed without merging**